src/reducers/InputOutput.js

```diff
@@ -46,15 +69,39 @@ export function materialsRemove(state, action) {
 }
 
 export function materialsExport(state, action) {
+    // TODO: GENERALIZE INDEX
+    const index = state.index || 0;
+    const stateItems = state.materials.slice();
+    const material = stateItems[index];
+    const newMaterial = new Material(material);
+    if (material.isNonPeriodic) {
+        const lattice = [material.Lattice.a, material.Lattice.b, material.Lattice.c];
+        const updatedBasis = new Made.Basis(material.basis);
+        const centerOfCoordiantes = updatedBasis.centerOfCoordinatesPoint;
+        const centerOfLattice = math.multiply(0.5, lattice);
+        const translationVector = math.subtract(centerOfLattice, centerOfCoordiantes);
+        updatedBasis.translateByVector(translationVector);
+        const newBasis = {
+            cordinates: updatedBasis._coordinates.array,
+            elements: updatedBasis._elements.array
+        }
+        newMaterial.isUpdated = material.isUpdated;
+        newMaterial.lattice = material.lattice;
+        newMaterial.isNonPeriodic = material.isNonPeriodic;
+        newMaterial.basis = newBasis;
+        state.materials[state.index] = newMaterial;
```

Comment:
The newMaterial has a defect currently. The value of Basis gets set to the default parameters of [Si], [0,0,0] instead of the values from basis. Need to figure out why that is happening